modules/apps/29-fee/keeper/escrow.go

```diff
@@ -26,64 +26,63 @@ func (k Keeper) EscrowPacketFee(ctx sdk.Context, identifiedFee types.IdentifiedP
 		return sdkerrors.Wrapf(types.ErrRefundAccNotFound, "account with address: %s not found", refundAcc)
 	}
 
-	coins := identifiedFee.Fee.RecvFee
-	coins = coins.Add(identifiedFee.Fee.AckFee...)
-	coins = coins.Add(identifiedFee.Fee.TimeoutFee...)
-
-	if err := k.bankKeeper.SendCoinsFromAccountToModule(
-		ctx, refundAcc, types.ModuleName, coins,
-	); err != nil {
+	coins := identifiedFee.Fee.Total()
+	if err := k.bankKeeper.SendCoinsFromAccountToModule(ctx, refundAcc, types.ModuleName, coins); err != nil {
 		return err
 	}
 
-	// Store fee in state for reference later
-	k.SetFeeInEscrow(ctx, identifiedFee)
+	packetFees := []types.IdentifiedPacketFee{identifiedFee}
+	if feesInEscrow, found := k.GetFeesInEscrow(ctx, identifiedFee.PacketId); found {
+		packetFees = append(packetFees, feesInEscrow.PacketFees...)
+	}
+
+	identifiedFees := types.NewIdentifiedPacketFees(packetFees)
+	k.SetFeesInEscrow(ctx, identifiedFee.PacketId, identifiedFees)
+
 	return nil
 }
 
 // DistributePacketFees pays the acknowledgement fee & receive fee for a given packetId while refunding the timeout fee to the refund account associated with the Fee.
-func (k Keeper) DistributePacketFees(ctx sdk.Context, refundAcc, forwardRelayer string, reverseRelayer sdk.AccAddress, feeInEscrow types.IdentifiedPacketFee) {
-	// distribute fee for forward relaying
-	forward, err := sdk.AccAddressFromBech32(forwardRelayer)
-	if err == nil {
-		k.distributeFee(ctx, forward, feeInEscrow.Fee.RecvFee)
-	}
-
-	// distribute fee for reverse relaying
-	k.distributeFee(ctx, reverseRelayer, feeInEscrow.Fee.AckFee)
+func (k Keeper) DistributePacketFees(ctx sdk.Context, forwardRelayer string, reverseRelayer sdk.AccAddress, feesInEscrow []types.IdentifiedPacketFee) {
+	for _, packetFee := range feesInEscrow {
+		// distribute fee for forward relaying
+		forward, err := sdk.AccAddressFromBech32(forwardRelayer)
```

Comment:
Pull this line outside for loop

Reply:
Sure, nice catch 👍 Should an error be returned here if the `forwardRelayer` is not a valid bech32 address? Or should it just continue and pay out remaining fees? 

The existing code here only pays out `RecvFee` if the error *is* `nil`

Reply:
Continue regardless of error, since it is correct functioning if it is an empty address. Refund the `recv_fee` to refund address if forward relayer is not a valid address

Reply:
updated: https://github.com/cosmos/ibc-go/pull/915/commits/d4db040b3e948d907718f4edf7026a944ecc3dd3